pkg/kube/inject/app_probe.go

```diff
@@ -239,6 +262,112 @@ func DumpAppProbers(pod *corev1.Pod, targetPort int32) string {
 	return string(b)
 }
 
+// lookupNamedPort finds the port number for a named port in the pod containers
+// If containerName is specified, it will first look for the port in that container before checking other containers
+func lookupNamedPort(pod *corev1.Pod, name string, containerName string) int32 {
+	// First look in the specified container if provided
+	if containerName != "" {
+		// Check regular containers
+		for _, container := range pod.Spec.Containers {
+			if container.Name == containerName {
+				for _, port := range container.Ports {
+					if port.Name == name {
+						return port.ContainerPort
+					}
+				}
+				break // Found the container but no matching port
+			}
+		}
+
+		// Check init containers
+		for _, container := range pod.Spec.InitContainers {
+			if container.Name == containerName {
+				for _, port := range container.Ports {
+					if port.Name == name {
+						return port.ContainerPort
+					}
+				}
+				break // Found the container but no matching port
+			}
+		}
+	}
+
+	// Fall back to checking all containers if not found in the specified container
+	for _, container := range allContainers(pod) {
+		// Skip the container we already checked
+		if container.Name == containerName {
+			continue
+		}
+		for _, port := range container.Ports {
+			if port.Name == name {
+				return port.ContainerPort
+			}
+		}
+	}
+	return 0
+}
+
+func getIncludedPorts(pod *corev1.Pod) map[int32]bool {
+	// Get pod annotations
+	blankPorts := make(map[int32]bool)
+	annotations := pod.Annotations
+	if annotations == nil {
+		// If no annotations, include all ports by default
+		return blankPorts
+	}
+
+	includedPorts := make(map[int32]bool)
+
+	includeInboundPortsKey := annotation.SidecarTrafficIncludeInboundPorts.Name
+	// Check if includeInboundPorts annotation is present
+	if includePortsStr, exists := annotations[includeInboundPortsKey]; exists && includePortsStr != "" {
+		// If includeInboundPorts is specified, only include those ports
+		if includePortsStr == "*" {
+			// "*" means include all ports
+			return blankPorts
+		}
+
+		// Parse comma-separated list of ports
+		for _, portStr := range splitPorts(includePortsStr) {
+			if port, err := strconv.Atoi(strings.TrimSpace(portStr)); err == nil {
+				includedPorts[int32(port)] = true
+			} else {
+				log.Errorf("Failed to parse port %v from includeInboundPorts: %v", portStr, err)
+			}
+		}
+	} else {
+		excludeInboundPortsKey := annotation.SidecarTrafficExcludeInboundPorts.Name
+		// Then exclude ports specified in excludeInboundPorts
+		if excludePortsStr, exists := annotations[excludeInboundPortsKey]; exists && excludePortsStr != "" {
```

Comment:
I think this has to be outside the else at line 338 because otherwise it will never consider excluded ports if included ports exist.

Reply:
As per istio docs - 
<img width="929" height="305" alt="image" src="https://github.com/user-attachments/assets/b1de0c25-833e-4cee-bc02-40e0bb921768" />

I have updated the code to handle it as per this logic.